src/integTest/java/com/unosquare/cvgenerator/controller/TemplateControllerIntegrationTest.java

```diff
@@ -0,0 +1,77 @@
+package com.unosquare.cvgenerator.controller;
+
+import com.flextrade.jfixture.JFixture;
+import com.unosquare.cvgenerator.AbstractWebIntegrationTest;
+import com.unosquare.cvgenerator.dao.TemplateRepository;
+import com.unosquare.cvgenerator.model.entity.TemplateDetail;
+import io.restassured.RestAssured;
+import org.apache.http.HttpStatus;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+
+import static org.hamcrest.CoreMatchers.equalTo;
+
+public class TemplateControllerIntegrationTest extends AbstractWebIntegrationTest {
+
+    @Autowired
+    private TemplateRepository repository;
```

Comment:
I think we'd be better following one convention for wiring up. You could add the requiredargsconstructor here and make it final.

Reply:
dunno if this works for test classes (need to try it out), usually it complains about needing a noargs constructor, so would have to set up both constructors

Reply:
have it feeling it might lead to repository not being autowired at all

Reply:
So the IOC framework is controlling the instantiation and the Template repository is still being managed. Really the only difference is constructor based vs property based injection. I'd be suprised if it didn't work honestly.